File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        if(!root) return 0;
        queue<TreeNode*> q;
        q.push(root);
        int maxLvl = 1, maxSum = INT_MIN, lvl = 1;
        while(!q.empty()) {
            int lvlSum = 0, lvlSize = q.size();;
            for (int i = 0; i < lvlSize; i++) {
                TreeNode* node = q.front();
                q.pop();
                
                lvlSum += node->val;

                if(lvlSum)
                if(node->left) q.push(node->left);
                if(node->right) q.push(node->right);
            } 

            if(lvlSum > maxSum) {
                maxSum = lvlSum;
                maxLvl = lvl;
            }

            lvl++;
        }

        return maxLvl;
    }
};

File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxLevelSum(TreeNode root) {
        if(root == null) return 0;

        Queue<TreeNode> q = new LinkedList<>();
        int maxLvlSum = Integer.MIN_VALUE, res = 1, lvl = 1;

        q.offer(root);

        while(!q.isEmpty()) {
            int currLvlSum = 0, size = q.size();
            for(int i = 0; i < size; i++) {
                TreeNode node = q.poll();
                currLvlSum += node.val;
                if(node.left != null) q.offer(node.left);
                if(node.right != null) q.offer(node.right); 
            }
            if(currLvlSum > maxLvlSum) {
                maxLvlSum = currLvlSum;
                res = lvl;
            }
            lvl++;
        }

        return res;
    }
}